lab-beeksma/model/storage.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const storage = {};
+
+module.exports = exports = {};
+
+exports.createItem = function(schemaName, item) {
+  if (!schemaName) return Promise.reject(new Error('expected schemaName'));
+  if (!item) return Promise.reject(new Error('expected item'));
+  if (!storage[schemaName]) storage[schemaName] = {};
+
+  storage[schemaName][item.id] = item;
+
+  return Promise.resolve(item);
+};
+
+exports.fetchItem = function(schemaName, id) {
+  return new Promise((resolve, reject) => {
+    if (!schemaName) return reject(new Error('expected schema name'));
+
+    var schema = storage[schemaName];
+    if (!schema) return reject(new Error('schema not found'));
+
+    if (!id){
+      var idArray = [];
+      for (var x in schema){
+        idArray.push(schema[x].id);
+      }
+      resolve({ids: idArray});
+    }
+    else{
+      var item = schema[id];
+      if (!item) return reject(new Error('item not found'));
+      resolve(item);
+    }
+  });
+};
+
+exports.removeItem = function(schemaName, id) {
+  return new Promise((resolve, reject) => {
+    if (!schemaName) return reject(new Error('expected schema name'));
+    if (!id) return reject(new Error('expected id'));
+
+    var schema = storage[schemaName];
+    if (!schema) return reject(new Error('schema not found'));
+
+    var item = schema[id];
+    if (!item) return reject(new Error('item not found'));
+
+    delete storage[schemaName][id];
+
+    resolve('Deleted');
```

Comment:
It can be useful on a delete to return the deleted item. Less useful to return a "status" string; better to just `resolve()` if the result isn't useful.